iocore/net/UnixUDPNet.cc

```diff
@@ -831,44 +834,68 @@ UDPNetHandler::mainNetEvent(int event, Event *e)
   (void)event;
   (void)e;
 
-  PollCont *pc = get_UDPPollCont(e->ethread);
+  UnixUDPConnection *uc;
+
+  // handle new UDP connection
+  SList(UnixUDPConnection, newconn_alink) ncq(newconn_list.popall());
+  while ((uc = ncq.pop())) {
+    if (uc->shouldDestroy()) {
+      uc->Release();
+    } else {
+      ink_assert(uc->mutex && uc->continuation);
+      open_list.enqueue(uc);
+    }
+  }
 
   // handle UDP outgoing engine
   udpOutQueue.service(this);
 
   // handle UDP read operations
-  UnixUDPConnection *uc, *next;
-  int i;
-  int nread = 0;
-
-  EventIO *temp_eptr = nullptr;
+  int i, nread = 0;
+  PollCont *pc = get_UDPPollCont(e->ethread);
+  EventIO *epd = nullptr;
   for (i = 0; i < pc->pollDescriptor->result; i++) {
-    temp_eptr = (EventIO *)get_ev_data(pc->pollDescriptor, i);
-    if ((get_ev_events(pc->pollDescriptor, i) & EVENTIO_READ) && temp_eptr->type == EVENTIO_UDP_CONNECTION) {
-      uc = temp_eptr->data.uc;
-      ink_assert(uc && uc->mutex && uc->continuation);
-      ink_assert(uc->refcount >= 1);
-      if (uc->shouldDestroy()) {
-        // udp_polling->remove(uc,uc->polling_link);
-        uc->Release();
+    epd = (EventIO *)get_ev_data(pc->pollDescriptor, i);
+    if (epd->type == EVENTIO_UDP_CONNECTION) {
+      // TODO: handle EVENTIO_ERROR
+      if (get_ev_events(pc->pollDescriptor, i) & EVENTIO_READ) {
+        uc = epd->data.uc;
+        ink_assert(uc && uc->mutex && uc->continuation);
+        ink_assert(uc->refcount >= 1);
+        if (uc->shouldDestroy()) {
+          open_list.remove(uc);
+          uc->Release();
+        } else {
+          udpNetInternal.udp_read_from_net(this, uc);
+          nread++;
+        }
       } else {
-        udpNetInternal.udp_read_from_net(this, uc);
-        nread++;
+        Debug("iocore_udp_main", "Unhandled epoll event: 0x%04x", get_ev_events(pc->pollDescriptor, i));
       }
-    } // if EPOLLIN
-  }   // end for
+    } else if (epd->type == EVENTIO_DNS_CONNECTION) {
+      // TODO: handle DNS conn if there is ET_UDP
```

Comment:
Do we really handle DNS connection here? It looks like DNS connection is on only ET_DNS or ET_CALL.

Reply:
IMO, It is better to handler the UDP packet by ET_UDP and handle the TCP connection by ET_NET.
It is a temp solution to handle the `DNSConnection` by ET_NET or ET_DNS thread group.

DNSHandler is probably the first one to utilize both the ET_UDP and ET_NET because there are DNS UDP and DNS TCP protocols.